RecoTracker/PixelTrackFitting/plugins/PixelTrackProducerFromSoA.cc

```diff
@@ -175,8 +175,12 @@ void PixelTrackProducerFromSoAT<TrackerTraits>::produce(edm::StreamID streamID,
   //sort index by pt
   std::vector<int32_t> sortIdxs(nTracks);
   std::iota(sortIdxs.begin(), sortIdxs.end(), 0);
+  //sort good-quality tracks by pt, keep bad-quality tracks in the bottom
   std::sort(sortIdxs.begin(), sortIdxs.end(), [&](int32_t const i1, int32_t const i2) {
-    return tsoa.view()[i1].pt() > tsoa.view()[i2].pt();
+    if (quality[i1] >= minQuality_ && quality[i2] >= minQuality_)
+      return tsoa.view()[i1].pt() > tsoa.view()[i2].pt();
+    else
+      return quality[i1] > quality[i2];
```

Comment:
without thinking about this a lot more, I'm not 100% sure this forms a 'stable sort'. For all possible values, does this guarantee 'if A < B and B < C then A < C'?

Reply:
Is it difficult to scan the pT and quality in a few events with bad quality tracks just to confirm the sort behaves as expected?

Reply:
I checked that the persistent tracks in `storeTracks` are sorted by pT, as expected, which is the behavior of the current code. I can also check the transient track collection.
Anyway, the main point was to not get accessed to `pt()` for tracks with bad quality, which is exactly what this PR is doing